packages/core/src/compiler/Style.test.ts

```diff
@@ -1328,4 +1328,25 @@ delete x.z.p }`,
       }
     });
   });
+
+  describe("gather dependencies", () => {
+    test("indexing", async () => {
+      const dsl = "type T";
+      const sub = "T t";
+      const sty =
+        canvasPreamble +
+        `
+forall T t {
+  t.xs = [1, 2, 3, 4, 5, 6]
+  t.index = 3
+  t.v = t.xs[t.index]
+}
+      `;
+
+      const { graph } = await loadProgs({ dsl, sub, sty });
+      expect(graph.parents("`t`.v").sort()).toEqual(
+        ["`t`.xs", "`t`.index"].sort()
+      );
```

Comment:
Thanks for adding this test! This isn't really what I meant, though; could you add a test that demonstrates a Style program that should succeed, but failed before this PR? You can use this same Style program you've already written here if you want, but test it in such a way that you're actually checking that running it works, rather than just looking at some intermediate state of the compiler.

Reply:
Sounds good - I'll see if I can add this test tonight.